### PR TITLE
ci: Only run doc-tests for library targets

### PR DESCRIPTION
**Pull request by emgrav**

### CODE REVIEW

File: .github/workflows/rust-workflow.yml [hunk 1]
```diff
@@ -49,7 +49,8 @@ jobs:
     
     - name: Doc-test
       working-directory: head
-      run: cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
+      # Don't run doc-tests for binary crates
+      run: cargo metadata --format-version 1 --no-deps | jq '.packages[].targets[].crate_types[] | select(. == "lib")' -e && cargo +${NIGHTLY_VERSION} test --doc --workspace --verbose
```
Comment (FlixCoder) [inline .github/workflows/rust-workflow.yml:53]: Is it possible to do `cargo +nightly test --doc --lib` to still test the library part of the binaries? (and to have an easier to understand command)

Comment (emgrav) [reply .github/workflows/rust-workflow.yml:53]: Afraid not.
```shell
$ cargo +nightly test --doc --lib
error: Can't mix --doc with other target selecting options
```

Comment (FlixCoder) [reply .github/workflows/rust-workflow.yml:53]: sad

Comment (FlixCoder) [inline .github/workflows/rust-workflow.yml:53]: Doesn't this make CI fail when it is binary? Did you test this?

**Comment by emgrav** [comment]
Closed, as any crates that run into this issue should be refactored to have a lib.rs anyway.

**Pull request closed without merging**